MeretzLib/Apple/iOS/MeretzSampleGame/MeretzSampleGame/GameViewController.m

```diff
@@ -8,351 +8,766 @@
+	// called when a vendorConsumeGetNew task finishes
+	// if successful (result.Success==TRUE), result.Items contains any newly acquired (un-consumed) items.
+	-(void) didVendorConsumeGetNewFinish:(MeretzVendorConsumeResult *)result;
+	// called when a vendorConsumeAcknowledge task finishes
+	// if successful (result.Success==TRUE), result.Items contains the acknowledged (consumed) items.
+	-(void) didVendorConsumeAcknowledgeFinish:(MeretzVendorConsumeResult *)result;
+
+@end
+
 @implementation GameViewController
 
-Meretz *gMeretz= nil;
-MeretzSampleGameState gSampleGameState= Initial;
-MeretzTaskId gConnectUserTaskId= MERETZ_TASK_ID_INVALID;
-MeretzTaskId gVendorConsumeTaskId= MERETZ_TASK_ID_INVALID;
-MeretzTaskId gVendorDisconnectTaskId= MERETZ_TASK_ID_INVALID;
+	@synthesize MeretzAPI;
+	@synthesize SampleGameState;
+	@synthesize VendorConnectUserTaskId;
+	@synthesize VendorConsumeTaskId;
+	@synthesize VendorDisconnectUserTaskId;
+	@synthesize ConnectionCodeTextField;
+	@synthesize ConnectionCode;
 
-static void updateSamepleGameState(void)
-{
-	switch (gSampleGameState)
+	-(void) initializeMeretzDemo
 	{
-		case Initial:
+		// if we have previously saved a Meretz Access Token, use it
+		// your game will want to use a similar pattern
+		NSString *savedAccessToken= [self getSavedMeretzAccessToken];
+		if ((nil != savedAccessToken) && (0 < [savedAccessToken length]))
 		{
-			// initiate a user connection task
-			NSString *userConnectionCode= TEST_MERETZ_USER_CONNECTION_CODE;
-			NSString *vendorUserIdentifier= SAMPLE_GAME_USER_IDENTIFIER;
+			NSLog(@"Retrieved saved Meretz access token: %@", savedAccessToken);
+		}
+		
+		id useDelegate= USE_MERETZ_DELEGATE ? self : nil;
+		
+		// initialize Meretz lib
+		NSLog(@"Initializing Meretz");
+		self.MeretzAPI= [[Meretz alloc] init:savedAccessToken optionalDelegate:useDelegate];
+		NSAssert(nil != self.MeretzAPI, @"Failed to initialize MeretzLib!");
+		
+		// setup some demo app state
+		self.SampleGameState= Initial;
+		self.VendorConnectUserTaskId= MERETZ_TASK_ID_INVALID;
+		self.VendorConsumeTaskId= MERETZ_TASK_ID_INVALID;
+		self.VendorDisconnectUserTaskId= MERETZ_TASK_ID_INVALID;
+		
+		self.ConnectionCodeTextField= nil;
+		self.ConnectionCode= @"";
+		
+		if (FALSE)
+		{
+			// point Meretz at a custom dev server
+			[MeretzAPI setMeretzHostName:@"127.0.0.1"];
+			[MeretzAPI setMeretzPort:8000];
+			[MeretzAPI setMeretzProtocol: @"http"];
+			[MeretzAPI setMeretzAPIPath:@""];
+			NSLog(@"Meretz server set to: %@", [MeretzAPI getMeretzServerString]);
+		}
+		
+		[self initializeMeretzDemoUI];
+		
+		// run a basic update loop, to allow us to check async operations in Meretz-land
+		float intervalSeconds= 0.25f;
+		[NSTimer scheduledTimerWithTimeInterval:intervalSeconds target:self selector:@selector(sampleGameTick:) userInfo:nil repeats:YES];
+		
+		return;
+	}
+
+	-(void) initializeMeretzDemoUI
+	{
+		float width= self.view.frame.size.width, height= 30.0f;
+		
```

Comment:
New UI consists of a text field for connection code entry / connection status, and 2 buttons (Consume Items, Disconnect)